serve.go

```diff
@@ -14,19 +17,23 @@ import (
 // It returns an error if the server could not start (it could not bind to the port for example).
 // It also generates the OpenAPI spec and outputs it to a file, the UI, and a handler (if enabled).
 func (s *Server) Run() error {
+	if err := s.setupDefaultListener(); err != nil {
+		return fmt.Errorf("failed to start server: %w", err)
+	}
 	s.setup()
```

Comment:
The setup listener is now called before setup, at all place setup is called if I'm not wrong

What about calling it in setup, and update setup signature to return an error?

Reply:
Thank you! Two heads are better than one. I made some adjustments and have been running manual tests (locally only).